corehq/apps/data_analytics/gir_generator.py

```diff
@@ -53,9 +60,21 @@ def fetch_latest_doc(dom1_id, dom2_id):
             for monthspan in self.monthspan_list:
                 gir_dict = GIRTableGenerator.get_gir_dict_for_domain_and_monthspan(domain, monthspan)
                 rows.append(gir_dict)
-        GIRRow.objects.bulk_create(
-            [GIRRow(**gir_row) for gir_row in rows]
-        )
+
+        final_rows = []
+        for row_data in rows:
+            try:
+                gir_row = GIRRow(**row_data)
+                final_rows.append(gir_row)
+            except IntegrityError:
```

Comment:
Wouldn't this be raised only when it's attempted to be sent to the DB, ie in the `bulk_create` below?